### PR TITLE
[Wi-Fi] [SiWx917] Adding the bring up changes for new Wi-Fi SDK 917 NCP

### PR DESCRIPTION
**Pull request by shgutte**
Problem / Feature
Adding the bring-up changes for the new Wi-Fi SDK917 NCP

Change Overview
What's in this PR
Adding the bring-up changes for the new Wi-Fi SDK917 NCP related to board 4186C and 4197C board autogen file changes

Testing
917 NCP + 4186C
917 NCP + 4187C

### CODE REVIEW

File: matter/efr32/efr32mg24/BRD4186C/autogen/sl_component_catalog.h [hunk 1]
```diff
@@ -15,6 +14,33 @@
 #define SL_CATALOG_BLUETOOTH_RTOS_ADAPTATION_PRESENT
 #define SL_CATALOG_BLUETOOTH_PRESENT
 #define SL_CATALOG_BLUETOOTH_FEATURE_MULTIPROTOCOL_PRESENT
+#ifdef CHIP_917
+#define SL_CATALOG_MATTER_ACCESS_CONTROL_PRESENT
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg24/BRD4186C/autogen/sl_component_catalog.h:18]: Why are those needed ? those all all define tie to slc matter component for matter clusters. Which doesn't do anything in CSA SMG context. 

Is there another usage for those?

File: matter/efr32/efr32mg24/BRD4186C/autogen/sl_event_handler.c [hunk 2]
```diff
@@ -91,6 +99,9 @@ void sl_service_init(void)
   sl_sleeptimer_init();
   sl_hfxo_manager_init();
   sl_mbedtls_init();
+#ifdef CHIP_917
+  sl_mpu_disable_execute_from_ram();
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg24/BRD4186C/autogen/sl_event_handler.c:103]: mpu is not enable in smg/csa. I  doubt you need this

File: matter/efr32/efr32mg24/BRD4186C/autogen/sl_event_handler.h [hunk 3]
```diff
@@ -7,5 +7,8 @@ void sl_driver_init(void);
 void sl_service_init(void);
 void sl_stack_init(void);
 void sl_internal_app_init(void);
+#ifdef CHIP_917
+void sl_iostream_init_instances(void);
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg24/BRD4186C/autogen/sl_event_handler.h:11]: don't you need this for CPMS re-provisionning ?

File: matter/efr32/efr32mg24/BRD4187C/autogen/sl_component_catalog.h [hunk 4]
```diff
@@ -14,6 +14,33 @@
 #define SL_CATALOG_BLUETOOTH_RTOS_ADAPTATION_PRESENT
 #define SL_CATALOG_BLUETOOTH_PRESENT
 #define SL_CATALOG_BLUETOOTH_FEATURE_MULTIPROTOCOL_PRESENT
+#ifdef CHIP_917
+#define SL_CATALOG_MATTER_ACCESS_CONTROL_PRESENT
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg24/BRD4187C/autogen/sl_component_catalog.h:18]: same comment has above

File: matter/efr32/efr32mg24/BRD4187C/autogen/sl_event_handler.c [hunk 5]
```diff
@@ -88,6 +99,9 @@ void sl_service_init(void)
   sl_sleeptimer_init();
   sl_hfxo_manager_init();
   sl_mbedtls_init();
+#ifdef CHIP_917
+  sl_mpu_disable_execute_from_ram();
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg24/BRD4187C/autogen/sl_event_handler.c:103]: same comment as previously

File: matter/efr32/efr32mg24/BRD4186C/autogen/sl_component_catalog.h (hunk 1 above)
Comment (shgutte) [reply matter/efr32/efr32mg24/BRD4186C/autogen/sl_component_catalog.h]: removed

File: matter/efr32/efr32mg24/BRD4187C/autogen/sl_component_catalog.h (hunk 4 above)
Comment (shgutte) [reply matter/efr32/efr32mg24/BRD4187C/autogen/sl_component_catalog.h]: removed

File: matter/efr32/efr32mg24/BRD4186C/autogen/sl_event_handler.c (hunk 2 above)
Comment (shgutte) [reply matter/efr32/efr32mg24/BRD4186C/autogen/sl_event_handler.c]: Removed

File: matter/efr32/efr32mg24/BRD4187C/autogen/sl_event_handler.c (hunk 5 above)
Comment (shgutte) [reply matter/efr32/efr32mg24/BRD4187C/autogen/sl_event_handler.c]: Removed

File: matter/efr32/efr32mg24/BRD4186C/autogen/sl_event_handler.c [hunk 6]
```diff
@@ -56,6 +59,9 @@ void sl_platform_init(void)
   sl_device_init_clocks();
   sl_device_init_emu();
   sl_board_init();
+#ifdef CHIP_917
+  bootloader_init();
```
[the remark this answers is not in the document]
Comment (shgutte) [reply matter/efr32/efr32mg24/BRD4186C/autogen/sl_event_handler.c]: removed

File: matter/efr32/efr32mg24/BRD4187C/autogen/sl_event_handler.h [hunk 7]
```diff
@@ -7,5 +7,8 @@ void sl_driver_init(void);
 void sl_service_init(void);
 void sl_stack_init(void);
 void sl_internal_app_init(void);
+#ifdef CHIP_917
+void sl_iostream_init_instances(void);
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg24/BRD4187C/autogen/sl_event_handler.h:11]: BTW I took it  out from here because the prototype is also defined in `autogen/sl_iostream_init_instances.h. ` Thats a SLC issue you can remove this one

File: matter/efr32/efr32mg24/BRD4187C/autogen/sl_event_handler.c [hunk 8]
```diff
@@ -31,13 +31,19 @@
 #ifdef SL_WIFI
 #include "sl_spidrv_instances.h"
 #endif
+#ifdef CHIP_917
+#include "btl_interface.h"
+#include "sl_mpu.h"
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg24/BRD4187C/autogen/sl_event_handler.c:36]: I believe you don't need either of those and neither the bootloader_init().

It will result in a cleaner file with less #ifdef if you remove it

File: matter/efr32/efr32mg24/BRD4186C/autogen/sl_event_handler.c [hunk 9]
```diff
@@ -31,7 +31,10 @@
 #ifdef SL_WIFI
 #include "sl_spidrv_instances.h"
 #endif
-
+#ifdef CHIP_917
+#include "btl_interface.h"
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg24/BRD4186C/autogen/sl_event_handler.c:35]: you removed the init call but not the include. You can remove those two also

File: matter/efr32/efr32mg24/BRD4187C/autogen/sl_event_handler.h (hunk 7 above)
Comment (shgutte) [reply matter/efr32/efr32mg24/BRD4187C/autogen/sl_event_handler.h]: removed

File: matter/efr32/efr32mg24/BRD4187C/autogen/sl_event_handler.c (hunk 8 above)
Comment (shgutte) [reply matter/efr32/efr32mg24/BRD4187C/autogen/sl_event_handler.c]: Removed

File: matter/efr32/efr32mg24/BRD4186C/autogen/sl_event_handler.c (hunk 9 above)
Comment (shgutte) [reply matter/efr32/efr32mg24/BRD4186C/autogen/sl_event_handler.c]: Removed

**Review by jmartinez-silabs — APPROVED** [review]
